nntrainer/compiler/onnx_interpreter.cpp

```diff
@@ -12,35 +12,249 @@
+    if (attr.name() == attr_name) {
+      std::ostringstream oss;
+      oss << prefix << "=";
 
-GraphRepresentation ONNXInterpreter::deserialize(const std::string &in) {
-  // Load and parse onnx file with protobuf
-  std::ifstream file(in, std::ios::binary);
-  if (!file.is_open()) {
-    throw std::runtime_error("File does not exist: " + in);
+      switch (attr.type()) {
+      case onnx::AttributeProto::INT:
+        oss << attr.i();
+        break;
+      case onnx::AttributeProto::INTS:
+        for (int i = start_offset; i < attr.ints_size(); ++i) {
+          oss << attr.ints(i);
+          if (i < attr.ints_size() - 1)
+            oss << separator;
+        }
+        break;
+      case onnx::AttributeProto::FLOAT:
+        oss << attr.f();
+        break;
+      case onnx::AttributeProto::STRING:
+        oss << attr.s();
+        break;
+      default:
+        throw std::runtime_error("Unsupported attribute type for " + attr_name);
+      }
+      return oss.str();
+    }
   }
 
-  onnx_model.ParseFromIstream(&file);
+  if (need_default_value) {
+    std::ostringstream oss;
+    oss << prefix << "=" << default_value;
+    return oss.str();
+  } else {
+    throw std::runtime_error("There is no matching attribute for " +
+                             node.name());
+  }
+}
 
-  // Create nntrainer model instance
-  GraphRepresentation graph;
-  //   std::vector<std::shared_ptr<ml::train::Layer>> layers;
+std::string ONNXInterpreter::extractTensorAttribute(
+  const onnx::NodeProto &node, const std::string &attr_name,
+  const std::string &prefix, const std::string &separator,
+  const int start_offset, const int alpha) {
+  for (const auto &attr : node.attribute()) {
+    if (attr.name() == attr_name) {
+      std::ostringstream oss;
+      oss << prefix << "=";
+
+      if (attr.has_t()) {
+        const auto &shape_tensor = attr.t().raw_data();
+        const int64_t *vals =
+          reinterpret_cast<const int64_t *>(shape_tensor.data());
+        int size = shape_tensor.size() / sizeof(int64_t);
+
+        for (int i = start_offset; i < size; ++i) {
+          oss << vals[i] + alpha;
+          if (i < size - 1)
+            oss << separator;
+        }
+        return oss.str();
+      }
+    }
+  }
+  throw std::runtime_error("There is no matching attribute for " + node.name());
+}
+
+void ONNXInterpreter::handleUnaryOp(const onnx::NodeProto &node,
+                                    GraphRepresentation &representation,
+                                    const std::string &op_type,
+                                    std::vector<std::string> &props) {
+  std::vector<std::string> inputNames = createOutputRemap(node);
+  props.push_back("name=" + cleanName(node.name()));
+  if (op_type == "activation") {
+    props.push_back("activation=" + activationKeyMap[node.op_type()]);
+  }
+  props.push_back("input_layers=" + inputNames[0]);
+
+  representation.push_back(
+    createLayerNode(op_type, {props.begin(), props.end()}));
```

Comment:
It is OK for now. (Not this PR). But I think we don't need to create actual nntrainer layer node for now. Because we also need to consider the delegate, the created layer could be meanless at this converting pass.